install-aws-ofi-nccl.sh

```diff
@@ -1,5 +1,19 @@
 #!/bin/bash
 
+os_name="$(. /etc/os-release; echo $NAME)"
+if [  "$os_name" == "Ubuntu" ]; then
```

Comment:
So we want to split the dependency installation to each software installation scripts? (same for all the other scripts).

Reply:
Yes. I think each script should be able to work independently.

Reply:
I'd want a different function for this, but I don't think it can work since package names can be different, for example g++ vs gcc-c++. Seems a little awful that we have to repeat the if/else chain over and over.